build_tooling/transform_asv_results.py

```diff
@@ -0,0 +1,160 @@
+"""
+import pandas as pd
+from arcticdb.storage_fixtures.s3 import real_s3_from_environment_variables
+import json
+from pathlib import Path
+from arcticdb import Arctic
+from argparse import ArgumentParser
+
+
+def setup_machine_folder(json_data, machine_path):
+    machine_path.mkdir()
+    with open(machine_path / "machine.json", "w") as out:
+        machine_data = json_data["params"]
+        machine_data["version"] = 1
+        json.dump(machine_data, out, indent=4, default=str)
+
+
+def df_to_asv_json(results_df: pd.DataFrame):
+    new_df = results_df.copy()
+    new_df["date"] = (pd.to_datetime(results_df["date"]).astype(int) // 1000000).astype(
+        object
+    )
+    new_df["version"] = results_df["version"].astype(object)
+
+    metadata = {
+        "commit_hash": new_df["commit_hash"].iloc[0],
+        "env_name": new_df["env_name"].iloc[0],
+        "date": new_df["date"].iloc[0],
+        "params": eval(new_df["params"].iloc[0]),
+        "python": new_df["python"].iloc[0],
+        "requirements": eval(new_df["requirements"].iloc[0]),
+        "env_vars": eval(new_df["env_vars"].iloc[0]),
+        "result_columns": eval(new_df["result_columns"].iloc[0]),
+        "durations": eval(new_df["durations"].iloc[0]),
+        "version": new_df["version"].iloc[0],
+    }
+
+    json_data = {**metadata, "results": {}}
+    for _, row in new_df.iterrows():
+        test_name = row["test_name"]
+        res_data = eval(row["results"])
+
+        json_data["results"][test_name] = res_data
+    return json_data
+
+
+def asv_json_to_df(full_path: str) -> pd.DataFrame:
+    with open(full_path, "r") as f:
+        data = json.load(f)
+
+    results_list = []
+    for test_name, test_results in data["results"].items():
+        flattened_data = pd.json_normalize(
+            {"test_name": test_name, "results": str(test_results)}
+        )
+        flattened_data["commit_hash"] = data["commit_hash"]
+        flattened_data["env_name"] = data["env_name"]
+        flattened_data["date"] = data["date"]
+        flattened_data["params"] = str(data["params"])
+        flattened_data["python"] = data["python"]
+        flattened_data["requirements"] = str(data["requirements"])
+        flattened_data["env_vars"] = str(data["env_vars"])
+        flattened_data["result_columns"] = str(data["result_columns"])
+        flattened_data["durations"] = str(data["durations"])
+        flattened_data["version"] = data["version"]
+        results_list.append(flattened_data)
+
+    results = pd.concat(results_list, ignore_index=True)
+    results["date"] = pd.to_datetime(data["date"], unit="ms")
+    return results
+
+
+def get_results_lib(arcticdb_client_override, arcticdb_library):
+    if arcticdb_client_override is not None:
+        ac = Arctic(arcticdb_client_override)
+    else:
+        factory = real_s3_from_environment_variables(shared_path=True)
```

Comment:
Which s3 bucket are we using for this? Would be good to have a dedicated one for benchmarks (in the correct region!) in case we need to tear down the shared one or whatever.

Reply:
it is using the one we use for the ci_test - arcticdb-ci-test-bucket-02.
But i agree, it is better to have it in a separate one.